gprofiler/profilers/java.py

```diff
@@ -259,11 +284,107 @@ def make_application_metadata(self, process: Process) -> Dict[str, Any]:
         # that loads other libs.
         libjvm_elfid = get_mapped_dso_elf_id(process, "/libjvm")
 
-        metadata = {"java_version": version, "libjvm_elfid": libjvm_elfid, "jvm_flags": get_jvm_flags(process)}
+        metadata = {"java_version": version, "libjvm_elfid": libjvm_elfid, "jvm_flags": self.get_jvm_flags(process)}
 
         metadata.update(super().make_application_metadata(process))
         return metadata
 
+    @functools.lru_cache(maxsize=1024)
+    def get_jvm_flags(self, process: Process) -> Optional[List[tuple]]:
+        jvm_raw_flags = self.get_jvm_flags_raw(process)
+        if jvm_raw_flags is None:
+            return None
+
+        return [dataclasses.astuple(flag) for flag in jvm_raw_flags if self.filter_jvm_flags(flag)]
```

Comment:
since we send it as a json it might cause a problem with the DB, as json dicts don't guarantee order whereas json list do.
https://stackoverflow.com/questions/7214293/is-the-order-of-elements-in-a-json-list-preserved

Reply:
We just need to make sure to dump to JSON with `sort_keys=True`. Can you check with the backend team that they load the JSON sorted?

Another thing: Maybe we should return a dict of (flag name) -> (flag attributes) instead of a list of (flag name and attributes).

Reply:
I think its OK both ways, discussed it briefly with @Jongy f2f, but lets rethink this, what format do we prefer?